x-pack/plugins/endpoint/server/routes/resolver.ts

```diff
@@ -0,0 +1,130 @@
+      try {
+        const query = await getESChildrenQuery(endpointAppContext, entityID, paginationInfo);
+        log.debug(`resolver/children query: ${JSON.stringify(query)}`);
+        const response = (await context.core.elasticsearch.dataClient.callAsCurrentUser(
+          'search',
+          query
+        )) as SearchResponse<ResolverData>;
+
+        if (response.hits.hits.length === 0) {
+          return res.notFound({ body: 'Nodes not found' });
+        }
+        const handler = new ResolverSearchHandler(
+          context.core.elasticsearch.dataClient,
+          endpointAppContext,
+          paginationInfo,
+          getESChildrenCountQuery(entityID),
+          entityID
+        );
+        return res.ok({
+          body: await handler.buildChildrenResponse(
+            response.hits.hits,
+            // total is an object in kibana >=7.0 so cast it to an object
+            // see https://github.com/elastic/kibana/issues/56694
+            (response.hits.total as unknown) as Total
+          ),
+        });
+      } catch (err) {
+        log.warn(err);
+        if (EntityParseError.isEntityParseError(err)) {
+          return res.badRequest({ body: err });
+        }
+        return res.internalError({ body: err });
+      }
+    }
+  );
+
+  router.get(
+    {
+      path: '/api/endpoint/resolver/node',
+      validate: {
+        query: validateQueryObject,
+      },
+      options: { authRequired: true },
+    },
+    async (context, req, res) => {
+      const entityID = req.query.entity_id;
+      log.debug(`entity_id: ${entityID}`);
+      const paginationInfo = {
+        page: req.query?.page_index,
+        pageSize: req.query?.page_size,
+      };
+      try {
+        const query = await getESNodeQuery(endpointAppContext, entityID, paginationInfo);
+        log.debug(`resolver/node query: ${JSON.stringify(query)}`);
+        const response = (await context.core.elasticsearch.dataClient.callAsCurrentUser(
+          'search',
+          query
+        )) as SearchResponse<ResolverData>;
+        if (response.hits.hits.length === 0) {
+          return res.notFound({ body: 'Node not found' });
+        }
+        const handler = new ResolverSearchHandler(
+          context.core.elasticsearch.dataClient,
+          endpointAppContext,
+          paginationInfo,
+          getESNodeCountQuery(entityID),
+          entityID
+        );
+        return res.ok({
+          body: await handler.buildNodeResponse(
+            response.hits.hits,
+            // total is an object in kibana >=7.0 so cast it to an object
+            // see https://github.com/elastic/kibana/issues/56694
+            (response.hits.total as unknown) as Total
+          ),
+        });
+      } catch (err) {
+        log.warn(err);
+        if (EntityParseError.isEntityParseError(err)) {
+          return res.badRequest({ body: err });
+        }
+        return res.internalError({ body: err });
+      }
+    }
+  );
+}
```

Comment:
looks like a lot of copy/pasted code between these two routes. Any plans on factoring some of it out?

Reply:
hmm are you referring to the error handling like this code:
```
        log.warn(err);
        if (EntityParseError.isEntityParseError(err)) {
          return res.badRequest({ body: err });
        }
        return res.internalError({ body: err });
```
I can refactor that. The other places are making slightly different calls when creating the `query` variable (`getESNodeQuery` vs `getESChildrenQuery`) and when building the `ResolverSearchHandler` object -> `getESChildrenCountQuery` vs `getESNodeCountQuery`